integration_test/run_tests.sh

```diff
@@ -1,47 +1,76 @@
-#!/bin/bash
-(
-  echo "##### Building SDK..." &&
-  (
-    cd ../ &&
-    rm -f firebase-functions-*.tgz &&
-    npm run build:pack &&
-    mv firebase-functions-*.tgz integration_test/functions/firebase-functions.tgz
-  ) &&
-  echo "##### Installing dependencies..." &&
-  (
-    cd functions &&
-    npm install
-  ) &&
-  echo "##### Deploying empty index.js to project..." &&
-  ./functions/node_modules/.bin/tsc -p functions/ &&  # Make sure the functions/lib directory actually exists.
-  echo "" > functions/lib/index.js &&
-  firebase deploy --debug 2> /dev/null &&
-  echo &&
-  echo "##### Project emptied. Deploying functions..." &&
-  ./functions/node_modules/.bin/tsc -p functions/ &&
-  firebase deploy --only functions --debug 2> /dev/null &&
-  echo &&
-  echo "##### Running the integration tests..." &&
-  funcURL=$(tail -n 1 firebase-debug.log | awk '{ print $5 }') &&  # URL is printed on last line of log.
-  wget --content-on-error -qO- $funcURL &&
-  echo &&
-  echo "##### Integration test run passed!" &&
-  echo "##### Re-deploying the same functions, to make sure updates work..." &&
-  firebase deploy --only functions --debug 2> /dev/null &&
-  echo &&
-  echo "##### Running the integration tests..." &&
-  wget --content-on-error -qO- $funcURL &&
-  echo &&
-  echo "##### Integration test run passed!" &&
-  echo "##### Removing all functions" &&
-  echo "" > functions/lib/index.js &&
-  firebase deploy --only functions --debug 2> /dev/null &&
-  rm functions/firebase-functions.tgz &&
-  rm -f functions/firebase-debug.log &&
-  echo &&
-  echo "##### All tests pass!"
-) || (
-  echo &&
-  echo "XXXXX Something failed XXXXX" &&
-  false  # Finish with an error code.
-)
+#!/usr/bin/env bash
+
+# Exit immediately if a command exits with a non-zero status.
+set -e
+
+function usage {
+  echo "Usage: $0 <project_id>"
+  exit 1
+}
+
+# The first parameter is required and is the Firebase project id.
+if [[ $1 == "" ]]; then
+  usage
+fi
+PROJECT_ID=$1
+
+# Directory where this script lives.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+function announce {
+  echo -e "\n##### $1"
+}
+
+function build_sdk {
+  announce "Building SDK..."
+  cd $DIR/..
+  rm -f firebase-functions-*.tgz
+  npm run build:pack
+  mv firebase-functions-*.tgz integration_test/functions/firebase-functions.tgz
+}
+
+function install_deps {
+  announce "Installing dependencies..."
+  cd $DIR/functions
+  npm install
+}
+
+function delete_all_functions {
+  announce "Deploying empty index.js to project..."
+  cd $DIR
+  ./functions/node_modules/.bin/tsc -p functions/ # Make sure the functions/lib directory actually exists.
+  echo "" > functions/lib/index.js
+  firebase deploy --project=$PROJECT_ID --only functions
+  announce "Project emptied."
+}
+
+function deploy {
+  announce "Deploying functions..."
+  cd $DIR
+  ./functions/node_modules/.bin/tsc -p functions/
+  firebase deploy --project=$PROJECT_ID --only functions
```

Comment:
This does need "--debug 2> /dev/null" to generate the firebase-debug.log file. Else a clean user will get this error:

> ##### Running the integration tests...
> tail: cannot open ‘/usr/local/google/home/rjh/src/firebase-functions/integration_test/firebase-debug.log’ > for reading: No such file or directory
> curl: no URL specified!
> curl: try 'curl --help' or 'curl --manual' for more information

Reply:
Oh I see, since we're depending on scraping the debug log and usually it gets deleted.  The `2> /dev/null` is the unfortunate part because it suppresses deploy errors.  Without it, though, we get tons of debug lines spamming the output.

Reply:
In 1937e8e I removed the log scraping in favor of just manually constructing the URL.  I'm open to other suggestions, but this seemed like the least-bad approach.  Additionally, I'd argue it's maybe even less brittle than relying on a very specific debug output format of `firebase-tools`.  Lmk what you think.

Reply:
Thanks! This seems the least-bad approch.